urlaa914a1de2_xu_hao_erilex_tgz-pJ8-erilex_esl_ESLJ8/src/erilex/codegen/Invocation.java

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2009, 2010 Hao Xu
+ertranne@hotmail.com
+
+This file is part of EriLex.
+
+EriLex is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+EriLex is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with EriLex; if not, write to the Free Software
+Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package erilex.codegen;
+
+import java.io.IOException;
+import java.io.Writer;
+
+/**
+ *
+ * @author ertri
+ */
+public class Invocation implements IExpression {
+
+    IExpression source;
+    String methodname;
+    IExpression[] args;
+    Type[] typeArgs;
+    IExpression[] impArgs;
+
+    public Invocation(IExpression object, String methodname, IExpression... params) {
+        this.args = params;
+        this.source = object;
+        this.methodname = methodname;
+    }
+
+    public Invocation(IExpression source, String methodname, IExpression[] args, Type[] typeArgs, IExpression[] impArgs) {
+        this.source = source;
+        this.methodname = methodname;
+        this.args = args;
+        this.typeArgs = typeArgs;
+        this.impArgs = impArgs;
+    }
+
+//    public ObjectCreation(Type type, Expression... args) {
+//        this.type = type;
+//        this.args = args;
+//    }
+    @Override
+    public void codeGen(Writer w, String indent) throws IOException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 59-73, lines 75-85, lines 86-99 into separate methods.